addons/point_of_sale/static/src/app/screens/partner_list/partner_list.js

```diff
@@ -121,6 +121,18 @@ export class PartnerList extends Component {
                 return (a.name || "").localeCompare(b.name || "");
             });
         }
+
+        for(const partner of res) {
+            if(!partner.address) {
+                partner.address =
+                    (partner.street ? partner.street + ", " : "") +
+                    (partner.zip ? partner.zip + ", " : "") +
+                    (partner.city ? partner.city + ", " : "") +
+                    (partner.state_id ? partner.state_id.name + ", " : "") +
+                    (partner.country_id ? partner.country_id.name : "");
+            }
+        }
+
```

Comment:
In 17.0 the address was added only once, when the session is loaded in `add_partners()` function. https://github.com/odoo/odoo/blob/55d8434a8ca12d83c89e0301ce31dc54f6395d98/addons/point_of_sale/static/src/app/store/db.js#L441-L446

Although you do it in the same way, here the address gets recomputed each time we open the customer list popup. **If we only want to load it once** we could use the field `contact_address_complete` form `res_partner`. The idea would be to load the field when we load the session data.
https://github.com/odoo/odoo/blob/d3c241ebf27bf5a6e44125ca42428450ccb59a85/addons/point_of_sale/models/pos_session.py#L197-L204
And then to change `address` to `contact_address_complete` in `searchString` and in the view **partner_line.xml**.

Note: `contact_address_complete` has a slightly different format in the sens that it does not put a coma in between zip and city but IMO this is a minor detail.

Reply:
The field `contact_address_complete` is defined in the enterprise code so we can't use it because it won't be available for a community only use.